frontend/streamlit_app.py

```diff
@@ -90,7 +129,7 @@ def main() -> None:
         with st.chat_message("user"):
             st.markdown(user_input)
 
-        response, sources = response_generator(user_input)
+        (response, sources), response_time = response_generator(user_input)
```

Comment:
Not a big fan of this. Instead do `response_tuple, response_time = response_generator()`. And also add output validation before doing `response, sources = response_tuple`